Comet integration updates

DN6 opened the pull request:
<!--
Thank you 🙏 for submitting a Pull Request to an Ultralytics 🚀 repo! We want to make contributing as possible. A few tips to get you started:

- Search existing PRs to see if a similar contribution already exists.
- Link this PR to an open Issue to help us understand what bug fix or feature is being implemented.
- Sign the Ultralytics Contributor License Agreement (CLA) by writing the below statement in a PR comment:  
I have read the CLA Document and I sign the CLA

Please see our ✅ [Contributing Guide](https://github.com/ultralytics/ultralytics/blob/master/CONTRIBUTING.md) for more details.
-->
This PR:

Updates the Comet ML integration with YOLOv8 

This PR:

1. Adds `Comet from` source tracking to YOLOv8
2. Adds support for logging training and validation images and model checkpoints every `N` epochs during training
3. Adds support for logging interactive Confusion Matrix to Comet
4. Adds support for Online and Offline Experiment logging 
5. Updates  Comet UTM links in the main README. 




## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Updated integration with Comet in the YOLOv8 code base.

### 📊 Key Changes
- Updated Comet integration links and documentation for YOLOv8.
- Introduced new Comet logging features and environment variables for customization.
- Enhanced logging of model information, validation predictions, confusion matrix, and various plots to Comet during training.
- Implemented functions to adjust bounding box annotations to the original image scale.
- Added support for offline Comet experiments.
- Provided a structured way to handle multi-process limitations when creating Comet experiments.

### 🎯 Purpose & Impact
- **For Developers**: Helps track and visualize the model training process more effectively, especially with large-scale or distributed training setups.
- **For End-Users**: Provides clearer insights into model performance and improves the debugging process by visualizing predictions and metrics related to the trained models.
- Overall, the update leads to a more comprehensive and user-friendly model training experience with valuable insights for performance optimization. 🚀

DN6 committed:
updates to comet integration

DN6 committed:
add support for logging image predictions, and confusion matrix

DN6 committed:
add max image predictions

DN6 committed:
update comet utms

DN6 committed:
reset image prediction count when training ends

DN6 commented:
I have read the CLA Document and I sign the CLA

DN6 committed:
Merge branch 'main' into comet-integration-updates

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

DN6 commented:
@glenn-jocher @AyushExel Submitting an update for the Comet integration. Would appreciate it if I could get your review on it.

DN6 commented:
Hi @glenn-jocher. The update details are in the PR description. Would you like me to expand on the specifics?

glenn-jocher commented:
@DN6 thank you for letting me know that the update details are in the PR description. I will take a look at it and review your changes. If I have any questions or need additional information, I'll be sure to let you know.

glenn-jocher committed:
Update comet.py

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

glenn-jocher committed:
Update Comet UTM link in both places in English README

Linked https://bit.ly/yolov8-readme-comet to https://www.comet.com/site/lp/yolov5-with-comet/?utm_source=yolov8&utm_medium=partner&utm_content=github

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

glenn-jocher committed:
Update 2 Comet links in README.zh-CN.md

@DN6

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

glenn-jocher committed:
Merge branch 'updates' into comet-integration-updates

glenn-jocher commented:
@DN6 ok, I've reviewed and cleaned it up a bit: re-introduced our standard warning when loggers aren't initialized, created a new bitly link that points to your UTM at https://bit.ly/yolov8-readme-comet and updated all 4 link locations in English and Chinese READMEs.

glenn-jocher commented on ultralytics/yolo/utils/callbacks/comet.py:
```diff
@@ -10,41 +13,319 @@
+    return pred_metadata_map
+
+
+def _log_confusion_matrix(experiment, trainer, curr_step, curr_epoch):
```
@DN6 unless I'm missing something here, this Confusion matrix will be incorrect as you are sorting class names alphabetically when the validator confusion matrix has been created with a class index sorting. This means you are detaching the class names from their corresponding index.

glenn-jocher committed:
Cleanup

glenn-jocher commented:
@DN6 I've got an apology in order here! I accidentally closed this when I merged 8.0.62 yesterday as I didn't realize this was targeted to `updates` branch rather than main. I'll go ahead and create a new PR since it doesn't seem possible to reopen.
